op-batcher/batcher/channel.go

```diff
@@ -109,41 +109,29 @@ func (s *channel) Timeout() uint64 {
 
 // updateInclusionBlocks finds the first & last confirmed tx and saves its inclusion numbers
 func (s *channel) updateInclusionBlocks() {
-	if len(s.confirmedTransactions) == 0 || !s.confirmedTxUpdated {
+	if len(s.confirmedTransactions) == 0 {
 		return
 	}
-	// If there are confirmed transactions, find the first + last confirmed block numbers
-	min := uint64(math.MaxUint64)
-	max := uint64(0)
+	s.minInclusionBlock = uint64(math.MaxUint64)
+	s.maxInclusionBlock = uint64(0)
 	for _, inclusionBlock := range s.confirmedTransactions {
-		if inclusionBlock.Number < min {
-			min = inclusionBlock.Number
-		}
-		if inclusionBlock.Number > max {
-			max = inclusionBlock.Number
-		}
+		s.minInclusionBlock = min(s.minInclusionBlock, inclusionBlock.Number)
+		s.maxInclusionBlock = max(s.maxInclusionBlock, inclusionBlock.Number)
 	}
-	s.minInclusionBlock = min
-	s.maxInclusionBlock = max
-	s.confirmedTxUpdated = false
 }
 
 // pendingChannelIsTimedOut returns true if submitted channel has timed out.
 // A channel has timed out if the difference in L1 Inclusion blocks between
 // the first & last included block is greater than or equal to the channel timeout.
 func (s *channel) isTimedOut() bool {
-	// Update min/max inclusion blocks for timeout check
-	s.updateInclusionBlocks()
```

Comment:
Is it worth adding to the godoc above, that `s.updateInclusionBlocks()` should be called first? Or even panicking if we detect that hasn't happened? I worry that in a future refactor we will forget this is important and then have a bug where we don't realised when a channel times out.

Reply:
Thanks, your comment made me realize that we can remove `updateInclusionBlocks` altogether (and the loop within), given it's only run after a tx is confirmed